ocs_ci/ocs/ui/page_objects/bucket_tab_permissions.py

```diff
@@ -0,0 +1,692 @@
+            PolicyType.ALLOW_PUBLIC_READ: self._build_public_read_policy,
+            PolicyType.ALLOW_SPECIFIC_ACCOUNT: self._build_specific_account_policy,
+            PolicyType.ENFORCE_HTTPS: self._build_enforce_https_policy,
+            PolicyType.ALLOW_FOLDER_ACCESS: self._build_folder_access_policy,
+        }
+
+        try:
+            bucket_policy_generated = policy_builders[policy_type](config)
+        except KeyError:
+            raise ValueError(f"Unsupported policy type: {policy_type}")
+
+        return json.dumps(bucket_policy_generated, indent=2)
+
+    def set_policy_json_in_editor(self, policy_json: str) -> None:
+        """
+        Set the policy JSON content in the code editor.
+
+        Args:
+            policy_json (str): JSON string to set in the editor.
+
+        Raises:
+            NoSuchElementException: If UI elements are not found.
+        """
+        try:
+            self.do_click(
+                self.bucket_tab["policy_code_editor"],
+                enable_screenshot=False,
+                copy_dom=False,
+            )
+        except TimeoutException:
+            pass
+
+        self._set_content_via_javascript(policy_json)
+
+    def _set_content_via_javascript(self, content: str) -> None:
+        """
+        Set content using JavaScript with Monaco and textarea fallbacks.
+
+        Args:
+            content (str): Content to set in the editor.
+
+        Raises:
+            TimeoutException: If all fallback strategies fail.
+        """
+        js_code = """
+        // Try Monaco editor API first
+        if (window.monaco && window.monaco.editor) {
+            const editors = window.monaco.editor.getEditors();
+            if (editors.length > 0) {
+                const editor = editors[0];
+                editor.setValue(arguments[0]);
+                return 'monaco_success';
+            }
+        }
+
+        // Fallback to textarea manipulation
+        const textArea = document.querySelector('textarea.inputarea');
+        if (textArea) {
+            textArea.value = arguments[0];
+            textArea.dispatchEvent(new Event('input', { bubbles: true }));
+            return 'textarea_success';
+        }
+
+        return 'failed';
+        """
+
+        try:
+            result = self.driver.execute_script(js_code, content)
+            if result == "failed":
+                error_msg = (
+                    "Failed to set policy JSON in Monaco editor using JavaScript approach. "
+                    "Check if Monaco editor is properly loaded and accessible."
+                )
+                logger.error(error_msg)
+                raise PolicyEditorError(error_msg)
+
+        except WebDriverException as e:
+            error_msg = (
+                "Failed to set policy JSON in Monaco editor using JavaScript approach. "
+                "Check if Monaco editor is properly loaded and accessible."
+            )
+            logger.error(error_msg)
```

Comment:
logger.error(error_msg) is redundant if we are logging same message with PolicyEditorError exception